packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx

```diff
@@ -2,74 +2,93 @@ import css from '@styled-system/css';
 import { AxisBottom, TickRendererProps } from '@visx/axis';
 import { GridColumns } from '@visx/grid';
 import { Group } from '@visx/group';
+import { PatternLines } from '@visx/pattern';
 import { Bar } from '@visx/shape';
-import { Text } from '@visx/text';
+import { Text as VisxText } from '@visx/text';
 import { KeyboardEvent, memo, MouseEvent } from 'react';
 import styled from 'styled-components';
-import siteText from '~/locale/index';
+import { Box } from '~/components-styled/base';
+import { Text } from '~/components-styled/typography';
 import { colors } from '~/style/theme';
-import { NationalTestedPerAgeGroupValue } from '~/types/data.d';
 import { formatPercentage } from '~/utils/formatNumber';
 import { AgeDemographicCoordinates } from './age-demographic-coordinates';
-export const AGE_GROUP_TOOLTIP_WIDTH = 340;
+import { AgeDemographicChartText, AgeDemographicDefaultValue } from './types';
+import { formatAgeGroupRange } from './utils';
 
-const text = siteText.infected_age_groups;
+export const AGE_GROUP_TOOLTIP_WIDTH = 340;
 
-interface AgeDemographicChartProps {
-  coordinates: AgeDemographicCoordinates;
-  onMouseMoveBar: (
-    value: NationalTestedPerAgeGroupValue,
-    event: MouseEvent<SVGElement>
-  ) => void;
+interface AgeDemographicChartProps<T extends AgeDemographicDefaultValue> {
+  coordinates: AgeDemographicCoordinates<T>;
+  text: AgeDemographicChartText;
+  onMouseMoveBar: (value: T, event: MouseEvent<SVGElement>) => void;
   onMouseLeaveBar: () => void;
   onKeyInput: (event: KeyboardEvent<SVGElement>) => void;
+  metricProperty: keyof T;
+  visuallyMaxPercentage?: number;
```

Comment:
```suggestion
  visualMaxPercentage?: number;
```
I feel that is more correct English. Another suggestion is `displayMaxPercentage`

Reply:
how about `clipValueAt`?

Reply:
I'll go for `displayMaxPercentage`.